spark-commons/src/main/scala/za/co/absa/spark/commons/utils/SchemaUtils.scala

```diff
@@ -93,4 +93,22 @@ object SchemaUtils {
     }
   }
 
+  /**
+   * Separates the field name components of a fully qualified column name as they hierachy goes from root down to the
+   * deepest one.
+   * Trailing '.' is ignored, leading one not.
+   *
+   * @param columnName  A fully qualified column name
+   * @return Each level field name in sequence how they go from root to the lowest one
+   */
+  def splitPath(columnName: String): List[String] = {
```

Comment:
I thought returning `Vector[String]` here, as a generally more efficient data type than `List[String]`, but couple of places where this function is used expect `List` and so far no place that would benefit from `Vector` has been spotted.

Reply:
Tbh I expect path being too short to need such optimisations. List is fine to me.